File: back/src/main/java/com/projet/valatMoslehGrodet/dto/AccountCreationDTO.java
package com.projet.valatMoslehGrodet.dto;

import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class AccountCreationDTO {
    @NotNull
    private String firstName;
    @NotNull
    private String lastName;
    @NotNull
    private String password;
    @NotNull
    private String email;
    private AddressDTO address;
    private Date birthDate;

}

File: front/src/lib/api/models/AccountCreationDto.ts
import { AddressDto } from "./AddressDto";

export type AccountCreationDto = {
  /**
   * @type string
   */
  firstName: string;
  /**
   * @type string
   */
  lastName: string;
  /**
   * @type string
   */
  password: string;
  /**
   * @type string
   */
  email: string;
  address: AddressDto;
  /**
   * @type string, date-time
   */
  birthDate: string;
};
